pkg/service/provider.go

```diff
@@ -0,0 +1,27 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-redis/redis/v8"
+)
+
+type RedisBasedKeyProvider struct {
+	client *redis.Client
+}
+
+func NewRedisBasedKeyProvider(c *redis.Client) *RedisBasedKeyProvider {
+	return &RedisBasedKeyProvider{
+		client: c,
+	}
+}
+
+func (p *RedisBasedKeyProvider) GetSecret(key string) string {
+	fmt.Print("GetSecret CALLED", "\n")
+	secret, _ := p.client.Get(context.Background(), key).Result()
```

Comment:
I could suggest storing api key/secret in a hash key. it's very hard to remove an API key without a hash of some sort. it's possible to use a hash with a name of "livekit-keys", and storing api key as key of the hash, and secret in the value.. i.e. `HGET livekit-keys APIxxxxx`